### PR TITLE
Improve file and folder structure

### PR DESCRIPTION
**Pull request by Vaddo**
- Update file dependencies of the new structure
- Rename `_helpers.spec.scss` to `functions.spec.scss`
- Rename `_utilities.spec.scss` to `_mixins.spec.scss`
- Struct and move logic from `_helpers.scss` and `_utilities.scss` to `_mixins.scss` and `_functions.scss`
- Split logic of `_layout.scss` to files `_general.scss`, `_box.scss`, `_responsive.scss`
- Rename `core` folder to `lib`
- Improve documentation and update dependencies in `hill-api.scss` and `hill.scss`
- Remove `normalize.css` from package.json`
- Rename `_vars.scss` to `_config.scss`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,15 @@
 # CHANGELOG
 
 ## HEAD
+* Update file dependencies of the new structure
```
Comment (ixisio) [inline CHANGELOG.md]: `Update file dependencies of the new structure` Was bedeutet das genau?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
+* Rename `_helpers.spec.scss` to `functions.spec.scss`
+* Rename `_utilities.spec.scss` to `_mixins.spec.scss`
+* Struct and move logic from `_helpers.scss` and `_utilities.scss` to `_mixins.scss` and `_functions.scss`
```
Comment (ixisio) [inline CHANGELOG.md]: `Struct` :)

File: lib/_functions.scss [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+
+
+//
+// Returns the number without unit. E.g. 30px -> 30
+// @param  {String} $val
+// @return {Number}
+// @hint Borrowed from Zurb Foundation _globals.scss
+//
+@function stripUnit($val) {
```
Comment (ixisio) [inline lib/_functions.scss]: This Method should not be public cause we use it also internally, e.g in `_box.scss`!

File: lib/core/_general.scss [hunk 4]
```diff
@@ -0,0 +1,23 @@
+@import 'helpers';
+
+
+
```
Comment (ixisio) [inline lib/core/_general.scss]: Please remove multiple empty lines.

File: lib/core/_general.scss [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+            $multiplier: '';
+        }
+
+        [#{$hill-layout-attribute-name}*="space-top#{$multiplier}"]       { margin-top:    #{$hill-layout-space * $index}; }
+        [#{$hill-layout-attribute-name}*="space-right#{$multiplier}"]     { margin-right:  #{$hill-layout-space * $index}; }
+        [#{$hill-layout-attribute-name}*="space-bottom#{$multiplier}"]    { margin-bottom: #{$hill-layout-space * $index}; }
+        [#{$hill-layout-attribute-name}*="space-left#{$multiplier}"]      { margin-left:   #{$hill-layout-space * $index}; }
+    }
+
+    [#{$hill-layout-attribute-name}*="float-left"]     { @extend  %_floatLeft; }
```
Comment (ixisio) [inline lib/core/_general.scss]: Why we us placeholder for `float: left;`?

File: lib/core/_helpers.scss [hunk 6]
```diff
@@ -1,3 +1,27 @@
+@import '../config';
+
+
+
+//
+// Added to reduce css output
+//
+%_floatLeft {
```
Comment (ixisio) [inline lib/core/_helpers.scss]: Skip this one, IMO

File: lib/core/_helpers.scss [hunk 7]
```diff
@@ -1,3 +1,27 @@
...
+        content: " ";
+        display: table;
+    }
+    &:after {
+        clear: both;
+    }
+}
+
+
+
```
Comment (ixisio) [inline lib/core/_helpers.scss]: Please remove empty lines (please use 1 single newline convention for the entire project)..

File: CHANGELOG.md (hunk 1 above)
Comment (Vaddo) [inline CHANGELOG.md]: I remove this point because everybody knows that you need to update file refecences/dependencies, if you change the filename

File: lib/_functions.scss (hunk 3 above)
Comment (Vaddo) [inline lib/_functions.scss]: You are right. I undo this change as we discussed.

File: lib/core/_general.scss (hunk 4 above)
Comment (Vaddo) [inline lib/core/_general.scss]: +1

File: lib/core/_general.scss (hunk 5 above)
Comment (Vaddo) [inline lib/core/_general.scss]: Sometimes you need only floating in your layout structure. But i think it is very exotic. What do you think?

I use `float-left` and `float-right` in one of my projects to align e.g. buttons.

File: lib/core/_helpers.scss (hunk 6 above)
Comment (Vaddo) [inline lib/core/_helpers.scss]: `%_floatLeft` is not needed anymore. I remove it

File: lib/core/_helpers.scss (hunk 7 above)
Comment (Vaddo) [inline lib/core/_helpers.scss]: +1

File: lib/core/_helpers.scss [hunk 8]
```diff
@@ -212,3 +212,13 @@
...
 }
+
+//
+// Returns the number without unit. E.g. 30px -> 30
+// @param  {String} $val
+// @return {Number}
+// @hint Borrowed from Zurb Foundation _globals.scss
+//
+@function stripUnit($val) {
+  @return $val / ($val * 0 + 1);
```
Comment (ixisio) [inline lib/core/_helpers.scss]: 4 spaces indentation please

**Comment by ixisio** [comment]
@Vaddo LGTM :+1:
